### PR TITLE
ci: Cleanup vvl.yml

### PR DESCRIPTION
**Pull request by juan-lunarg**
- Enable linker warnings as errors on all major platforms
- Cleanup matrix config usage
- Only test CMake 3.17.2 on `linuxBuild`
- Remove names that aren't that helpful

### CODE REVIEW

File: .github/workflows/vvl.yml [hunk 1]
```diff
@@ -37,100 +37,82 @@ permissions:
...
-      CMAKE_C_COMPILER_LAUNCHER: ccache
-      CMAKE_CXX_COMPILER_LAUNCHER: ccache
     runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
         include:
           # Test Ubuntu-20.04 release build works.
           - os: ubuntu-20.04
-            compiler: {cc: gcc, cxx: g++}
```
Comment (juan-lunarg) [inline .github/workflows/vvl.yml:50]: The default compiler for linux is already gcc so this doesn't need to be specified.

File: .github/workflows/vvl.yml [hunk 2]
```diff
@@ -37,100 +37,82 @@ permissions:
...
             config: release
             cpp_std: 17
-            robin_hood: "ON"
```
Comment (juan-lunarg) [inline .github/workflows/vvl.yml:53]: We were only testing "ON" so this can be removed.

File: .github/workflows/vvl.yml [hunk 3]
```diff
@@ -37,100 +37,82 @@ permissions:
...
             config: debug
             cpp_std: 20
-            robin_hood: "ON"
     steps:
       - uses: actions/checkout@v3
-      - uses: lukka/get-cmake@latest
+      - name: Test Minimum CMake Version
+        uses: lukka/get-cmake@latest
         with:
           cmakeVersion: 3.17.2
```
Comment (juan-lunarg) [inline .github/workflows/vvl.yml:59]: Note all the other builds can use cmake latest. We just need to make sure at least one build is testing the current minimum 3.17.2 to ensure we don't break backcompat by accident.

**Review by spencer-lunarg — APPROVED** [review]

File: .github/workflows/vvl.yml [hunk 4]
```diff
@@ -179,11 +154,8 @@ jobs:
   windows:
     runs-on: windows-2022
     strategy:
-      fail-fast: false
```
Comment (juan-lunarg) [inline .github/workflows/vvl.yml:182]: I left this in when I was debugging a while ago.
